src/relatedContent.ts

```diff
@@ -0,0 +1,87 @@
+import { Content } from "@guardian/content-api-models/v1/content";
+import { RelatedItemType } from "@guardian/apps-rendering-api-models/relatedItemType";
+import { RelatedContent } from "@guardian/apps-rendering-api-models/relatedContent";
+import { Image } from '@guardian/apps-rendering-api-models/image';
+import { map, OptionKind, fromNullable } from "@guardian/types/option";
+import { isFeature, isReview, isAnalysis, articleMainImage } from "capi";
+import { isLive, isComment, isAudio, isVideo, isGallery, isAdvertisementFeature } from "item";
+import { pipe2, pipe } from "lib";
+
+const parseRelatedItemType = (content: Content): RelatedItemType => {
+	const { tags } = content;
+	if (isFeature(content)) {
+        return RelatedItemType.FEATURE
+	} else if (isLive(tags) && content.fields?.liveBloggingNow) {
+		return RelatedItemType.LIVE
+	} else if (isReview(content)) {
+		return RelatedItemType.REVIEW
+	} else if (isAnalysis(content)) {
+		return RelatedItemType.ANALYSIS
+	} else if (isComment(tags)) {
+		return RelatedItemType.COMMENT
+	} else if (isAudio(tags)) {
+		return RelatedItemType.AUDIO
+	} else if (isVideo(tags)) {
+		return RelatedItemType.VIDEO
+	} else if (isGallery(tags)) {
+		return RelatedItemType.GALLERY
+	} else if (isAdvertisementFeature(tags)) {
+		return RelatedItemType.ADVERTISEMENT_FEATURE
+	} else {
+		return RelatedItemType.ARTICLE
+	}
+}
+
+const parseHeaderImage = (content: Content): Image | undefined => {
+	const optionalImage = pipe(
+        articleMainImage(content),
+        map(element => {
```

Comment:
I think if you change this to an `andThen` you can avoid the nested `Option` below on line 54:

```suggestion
        andThen(element => {
```